File: database/src/main/java/com/pelisat/cesp/ceemsp/database/model/Vehiculo.java
package com.pelisat.cesp.ceemsp.database.model;

import com.pelisat.cesp.ceemsp.database.type.NivelBlindajeEnum;
import com.pelisat.cesp.ceemsp.database.type.VehiculoOrigenEnum;
import com.pelisat.cesp.ceemsp.database.type.VehiculoStatusEnum;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "VEHICULOS")
@Getter
@Setter
public class Vehiculo extends CommonModel {
    @Column(name = "EMPRESA", nullable = false)
    private int empresa;

    @Column(name = "TIPO", nullable = false)
    private int tipo;

    @Column(name = "MARCA", nullable = false)
    private int marca;

    @Column(name = "SUBMARCA", nullable = false)
    private int submarca;

    @Column(name = "ANIO", nullable = false)
    private String anio;

    @Column(name = "ROTULADO")
    private boolean rotulado;

    @Column(name = "PLACAS", nullable = false)
    private String placas;

    @Column(name = "SERIE", nullable = false)
    private String serie;

    @Column(name = "ORIGEN", nullable = false)
    @Enumerated(EnumType.STRING)
    private VehiculoOrigenEnum origen;

    @Column(name = "BLINDADO")
    private boolean blindado;

    @Column(name = "SERIE_BLINDAJE")
    private String serieBlindaje;

    @Column(name = "FECHA_BLINDAJE")
    private LocalDate fechaBlindaje;

    @Column(name = "NUMERO_HOLOGRAMA")
    private String numeroHolograma;

    @Column(name = "PLACA_METALICA")
    private String placaMetalica;

    @Column(name = "EMPRESA_BLINDAJE")
    private String empresaBlindaje;

    @Column(name = "NIVEL_BLINDAJE")
    @Enumerated(EnumType.STRING)
    private NivelBlindajeEnum nivelBlindaje;

    @Column(name = "USO", nullable = false)
    private int uso;

    @Column(name = "RAZON_SOCIAL")
    private String razonSocial;

    @Column(name = "FECHA_INICIO")
    private LocalDate fechaInicio;

    @Column(name = "FECHA_FIN")
    private LocalDate fechaFin;

    @Column(name = "MOTIVO_BAJA")
    private String motivoBaja;

    @Column(name = "OBSERVACIONES_BAJA")
    private String observacionesBaja;

    @Column(name = "DOCUMENTO_FUNDATORIO_BAJA")
    private String documentoFundatorioBaja;

    @Column(name = "DOMICILIO")
    private int domicilio;

    @Column(name = "CONSTANCIA_BLINDAJE")
    private String constanciaBlindaje;

    @Column(name = "PERSONAL_ASIGNADO")
    private Integer personalAsignado;

    @Column(name = "FECHA_BAJA")
    private LocalDate fechaBaja;

    @Column(name = "STATUS")
    @Enumerated(EnumType.STRING)
    private VehiculoStatusEnum status;

    @Column(name = "COLORES_CAPTURADO")
    private boolean coloresCapturado;

    @Column(name = "FOTOGRAFIA_CAPTURADA")
    private boolean fotografiaCapturada;

    public Vehiculo() {
        super();
    }
}

File: ceemsp-frontend/src/app/_models/Vehiculo.ts
import {Deserializable} from "./Deserializable";
import VehiculoTipo from "./VehiculoTipo";
import VehiculoMarca from "./VehiculoMarca";
import VehiculoSubmarca from "./VehiculoSubmarca";
import VehiculoColor from "./VehiculoColor";
import VehiculoFotografiaMetadata from "./VehiculoFotografiaMetadata";
import VehiculoUso from "./VehiculoUso";
import EmpresaDomicilio from "./EmpresaDomicilio";
import Persona from "./Persona";

export default class Vehiculo implements Deserializable {
  id: number;
  uuid: string;
  tipo: VehiculoTipo;
  marca: VehiculoMarca;
  submarca: VehiculoSubmarca;
  uso: VehiculoUso;
  anio: string;
  color: string;
  rotulado: boolean;
  placas: string;
  serie: string;
  origen: string;
  blindado: boolean;
  serieBlindaje: string;
  fechaBlindaje: string;
  numeroHolograma: string;
  placaMetalica: string;
  empresaBlindaje: string;
  nivelBlindaje: string;
  razonSocial: string;
  fechaInicio: string;
  fechaFin: string;
  personalAsignado: Persona;
  eliminado: boolean;

  motivoBaja: string;
  observacionesBaja: string;
  documentoFundatorioBaja: string;
  fechaBaja: string;

  colores: VehiculoColor[];
  fotografias: VehiculoFotografiaMetadata[];
  domicilio: EmpresaDomicilio;

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }

  static obtenerColumnasPorDefault() {
    return [
      {headerName: 'ID', field: 'uuid', sortable: true, filter: true},
      {headerName: 'Tipo', field: 'tipo.nombre', sortable: true, filter: true },
      {headerName: 'Marca', field: 'marca.nombre', sortable: true, filter: true},
      {headerName: 'Submarca', field: 'submarca.nombre', sortable: true, filter: true},
      {headerName: 'Placas', field: 'placas', sortable: true, filter: true},
      {headerName: 'Serie', field: 'serie', sortable: true, filter: true}
    ];
  }

  static obtenerTodasLasColumnas() {
    return [
      {headerName: 'ID', field: 'uuid', sortable: true, filter: true},
      {headerName: 'Tipo', field: 'tipo.nombre', sortable: true, filter: true },
      {headerName: 'Marca', field: 'marca.nombre', sortable: true, filter: true},
      {headerName: 'Submarca', field: 'submarca.nombre', sortable: true, filter: true},
      {headerName: 'Placas', field: 'placas', sortable: true, filter: true},
      {headerName: 'Anio', field: 'anio', sortable: true, filter: true},
      {headerName: 'Serie', field: 'serie', sortable: true, filter: true},
      {headerName: 'Rotulado', field: 'rotulado', sortable: true, filter: true},
      {headerName: 'Origen', field: 'origen', sortable: true, filter: true},
      {headerName: 'Blindado', field: 'blindado', sortable: true, filter: true},
      {headerName: 'Serie Blindaje', field: 'serieBlindaje', sortable: true, filter: true},
      {headerName: 'Fecha Blindaje', field: 'fechaBlindaje', sortable: true, filter: true},
      {headerName: 'Numero holograma', field: 'numeroHolograma', sortable: true, filter: true},
      {headerName: 'Placa metalica', field: 'placaMetalica', sortable: true, filter: true},
      {headerName: 'Empresa blindaje', field: 'empresaBlindaje', sortable: true, filter: true},
      {headerName: 'Nivel blindaje', field: 'nivelBlindaje', sortable: true, filter: true}
    ]
  }
}
